src/betterproto/__init__.py

```diff
@@ -1404,6 +1403,15 @@ def to_dict(
         Dict[:class:`str`, Any]
             The JSON serializable dict representation of this object.
         """
+        # Mirror of from_dict: Struct's `fields` member is transparently
+        # dispatched through instead.
+        if isinstance(self, Struct):
+            output = {**self.fields}
+            for k in self.fields:
+                if hasattr(self.fields[k], "to_dict"):
+                    output[k] = self.fields[k].to_dict(casing, include_default_values)
+            return output
```

Comment:
This special-casing is arguably pretty ugly, but I couldn't think of a better (clean) way to accomplish this: `Struct` *is* kind of special in protobuf, in that it's a transparent mapping type (when expressed as JSON) where no other types really are. As such, it doesn't really fit into the `to_dict`/`from_dict` behavior below.